data-processing-lib/ray/src/data_processing_ray/runtime/ray/ray_utils.py

```diff
@@ -109,16 +111,30 @@ def operator() -> ActorHandle:
             time.sleep(creation_delay)
             return clazz.options(**actor_options).remote(params)
 
+        logger = get_logger(__name__)
         cls_name = clazz.__class__.__name__.replace('ActorClass(', '').replace(')','')
+        current = list_actors(filters=[("class_name", "=", cls_name), ("state", "=", "ALIVE")], limit=MAX_LIST)
+        c_len = len(current)
         actors = [operator() for _ in range(n_actors)]
-        for i in range(120):
-            time.sleep(1)
-            alive = list_actors(filters=[("class_name", "=", cls_name), ("state", "=", "ALIVE")])
-            if len(actors) == len(alive):
-                return actors
-        # failed - raise an exception
-        print(f"created {actors}, alive {alive}")
-        raise UnrecoverableException(f"out of {len(actors)} created actors only {len(alive)} alive")
+        overall = c_len + n_actors
+        if overall < MAX_LIST:
```

Comment:
If the `overall` number of actors is less than `MAX_LIST`, we can get their state from the ray server. Otherwise, we cannot determine their state, and the code execution becomes unpredictable. Therefore, we throw an `UnrecoverableException` if we have more than 10000 actors of the same type instantiated.  As a reference, the largest transforms we run now, have about 400 actors of a given type.